### PR TITLE
Blacklist roles

### PR DESCRIPTION
**Pull request by scottbucher**
Introduces roles into the birthday blacklist system. Users with one or more roles that are blacklisted will not have their birthday celebrated no matter what.

Bug fixes & improvements:
- Fixed issue with blacklist while testing birthday events
- Added ability to remove user/role from blacklist using their id.

### CODE REVIEW

File: src/services/celebration-service.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class CelebrationService {
 
                     // Get the blacklist data for this guild
                     let blacklistData = guildCelebrationData.blacklistedMembers.map(
-                        m => m.UserDiscordId
+                        m => m.DiscordId
```
Comment (KevinNovak) [inline src/services/celebration-service.ts:59]: Similar to above, maybe rename `m` to `b`?

File: src/services/celebration-service.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ export class CelebrationService {
 
                     // Get the blacklist data for this guild
                     let blacklistData = guildCelebrationData.blacklistedMembers.map(
```
Comment (KevinNovak) [inline src/services/celebration-service.ts:58]: You might want to rename `blacklistedMembers` to just `blacklist` or something

File: src/services/celebration-service.ts [hunk 3]
```diff
@@ -66,14 +66,18 @@ export class CelebrationService {
...
 
-                    let membersWithBirthdayTodayOrTomorrow = guildMembers.filter(member =>
-                        CelebrationUtils.isBirthdayTodayOrYesterday(
-                            memberUserDatas.find(data => data.UserDiscordId === member.id),
-                            guildData
-                        )
+                    let membersWithBirthdayTodayOrYesterday = guildMembers.filter(
+                        member =>
+                            !member.roles.cache
+                                .map(r => r.id)
```
Comment (KevinNovak) [inline src/services/celebration-service.ts:72]: You should be able to do `member.roles.cache.keyArray()`

**Review by KevinNovak — CHANGES_REQUESTED** [review]

File: src/services/celebration-service.ts (hunk 3 above)
Comment (scottbucher) [reply src/services/celebration-service.ts:72]: keyArray() is no longer supported, I assume you mean `[...member.roles.cache.keys()]`?

File: src/services/celebration-service.ts (hunk 2 above)
Comment (scottbucher) [reply src/services/celebration-service.ts]: Done

File: src/services/celebration-service.ts (hunk 1 above)
Comment (scottbucher) [reply src/services/celebration-service.ts]: Done

**Review by KevinNovak — APPROVED** [review]

**Pull request merged**